tsfile/src/main/java/org/apache/iotdb/tsfile/read/expression/util/ExpressionOptimizer.java

```diff
@@ -108,13 +108,66 @@ private IExpression handleOneGlobalTimeFilter(GlobalTimeExpression globalTimeExp
       addTimeFilterToQueryFilter((globalTimeExpression).getFilter(), regularRightIExpression);
       return regularRightIExpression;
     } else if (relation == ExpressionType.OR) {
-      return BinaryExpression
-          .or(pushGlobalTimeFilterToAllSeries(globalTimeExpression, selectedSeries),
-              regularRightIExpression);
+      IExpression afterTransform = pushGlobalTimeFilterToAllSeries(globalTimeExpression,
+          selectedSeries);
+      return mergeSecondTreeToFirstTree(afterTransform, regularRightIExpression);
     }
     throw new QueryFilterOptimizationException("unknown relation in IExpression:" + relation);
   }
 
+  /**
+   * This method merge the second input, which is of tree structure, to the first parameter. It
+   * visits all leaf nodes, which are SingleSeriesExpressions, or AndExpression in right Expression,
+   * merge them to the right position in leftExpression.
+   *
+   * @param leftExpression  The IExpression transformed from GlobalTimeExpression, which might have
+   *                        already be updated and merged.
+   * @param rightExpression The IExpression to be merged into the first IExpression
+   * @return a merged IExpression, which is initially based on the input leftExpression
+   */
+  private IExpression mergeSecondTreeToFirstTree(IExpression leftExpression,
+      IExpression rightExpression) {
+    if (rightExpression.getType() == ExpressionType.SERIES) {
+      SingleSeriesExpression leaf = (SingleSeriesExpression) rightExpression;
+      updateFilterWithOr(leftExpression, leaf.getFilter(), leaf.getSeriesPath());
+      return leftExpression;
+    } else if (rightExpression.getType() == ExpressionType.SERIES) {
```

Comment:
Two of if are same, please check the logic

Reply:
It's a mistake. Thanks for your reminding.